prow/kube/helpers.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kube
+
+// LabelsMatch returns whether the two provided label sets match.
+// l1 needs to be a subset or match exactly l2.
+func LabelsMatch(l1, l2 map[string]string) bool {
```

Comment:
The first argument should be a selector that is immediately parsed.

Reply:
Yeah, that's was I was doing before in this helper. The problem with selector parsing is that if we are going to support the whole selector syntax (which is basically what I am doing in the controllers by passing the selector through to the API server) then we need to import some stuff from apimachinery (because there is a LOT of stuff required for guaranteeing that we are doing selector parsing correctly).

Reply:
Well we are still parsing the selector the way things are now (outside `LabelsMatch` instead of inside). I agree that we need more complete parsing, but that seems orthogonal to whether the `LabelMatch` function does the parse itself or requires its caller to do so.
Is importing the selector parsing going to require importing a bunch of other extraneous code? I know that was the concern with importing the k8s resource struct types from client-go. (ref #1986)

Reply:
The only parsing we are doing today is just for testing purposes, hence far from complete. When we import apimachinery/pkg/labels, this helper becomes obsolete.

> Is importing the selector parsing going to require importing a bunch of other extraneous code? I know that was the concern with importing the k8s resource struct types from client-go. (ref #1986)

I was planning on copying rather than vendoring. The new code should be self-contained with a couple of minor tweaks.